arpitkh96/AmazeFileManager
# Fix SSH with password authentication

## Description
- Fix SftpEntryDao SQL declarations. sqlite is case insensitive, but just to be sure
- Use Kotlin's null/empty check in SshAuthenticationTask and SshConnectionPool
- Trim trailing \n when encrypting SSH passwords
- returns SSH path with encrypted password out to Drawer when calling UtilsHandler.getSftpList()
- Fix handling of encrypted passwords at SshClientUtils.extractBaseUriFrom(), use our own parsing routine instead of Uri.parse which will break when encrypted password contains slash characters

#### Issue tracker   
Fixes #2843

#### Automatic tests
On handling of SSH URI with encrypted passwords
  
#### Manual tests
  
Device: Pixel 2 emulator
OS: Android 11

Manual tests include
- create, update, delete SSH connection settings with SSH server requiring password authentication
- create, update, delete SSH connection settings with SSH server requiring key authentication
- open connection to SSH server should continue function as normal
- passwords, even encrypted passwords, should never be visible at Amaze's bottom bar

#### Build tasks success  

Successfully running following tasks on local:
